src/proteus/dswx_hls.py

```diff
@@ -2734,6 +2785,175 @@ def _warp(input_file, geotransform, projection,
+              in km
+       scratch_dir: str
+              Directory for temporary files
+       geotransform: numpy.ndarray
+              Geotransform describing the DSWx-HLS product geolocation
+       projection: str
+              DSWx-HLS product's projection
+       length: int
+              DSWx-HLS product's length (number of lines)
+       width: int
+              DSWx-HLS product's width (number of columns)
+
+       Returns
+       -------
+       ocean_mask : numpy.ndarray
+              Ocean mask (1: land, 0: ocean)
+    """
+    logger.info('creating ocean mask')
+
+    tile_min_x_utm, tile_dx_utm, _, tile_max_y_utm, _, tile_dy_utm = \
+        geotransform
+    tile_max_x_utm = tile_min_x_utm + width * tile_dx_utm
+    tile_min_y_utm = tile_max_y_utm + length * tile_dy_utm
+
+    tile_srs = osr.SpatialReference()
+    if projection.upper() == 'WGS84':
+        tile_srs.SetWellKnownGeogCS(projection)
+    else:
+        tile_srs.ImportFromProj4(projection)
+
+    tile_polygon = None
+    ocean_mask = np.zeros((length, width), dtype=np.uint8)
+    shapefile_ds = ogr.Open(shape_file, 0)
+
+    for layer in shapefile_ds:
+        for feature in layer:
+            geom = feature.GetGeometryRef()
+            if geom.GetGeometryName() != 'POLYGON':
+                continue
+
+            if tile_polygon is None:
+                polygon_srs = geom.GetSpatialReference()
+                tile_polygon, tile_min_y, tile_max_y, tile_min_x, tile_max_x = \
+                    _get_tile_srs_bbox(tile_min_y_utm, tile_max_y_utm,
+                                       tile_min_x_utm, tile_max_x_utm,
+                                       tile_srs, polygon_srs)
+
+            min_x, max_x, min_y, max_y = geom.GetEnvelope()
+            if not (min_x < tile_min_x < max_x or
+                    min_x < tile_max_x < max_x):
+                # the tile does not intersect with feature (lon test)
+                continue
+            if not (min_y < tile_min_y < max_y or
+                    min_y < tile_max_y < max_y):
+                # the tile does not intersect with feature (lat test)
+                continue
+
+            # intersect shoreline polygon to the tile and update its
+            # spatial reference system (SRS) to match the tile SRS
+            intersection_polygon = geom.Intersection(tile_polygon)
+            intersection_polygon.AssignSpatialReference(polygon_srs)
+            intersection_polygon.TransformTo(tile_srs)
+
+            # add margin to polygon
+            margin_m = int(1000 * margin_km)
+            intersection_polygon = intersection_polygon.Buffer(margin_m)
+
+            # Update feature with intersected polygon
+            feature.SetGeometry(intersection_polygon)
+
+            # Set up the shapefile driver 
+            shapefile_driver = ogr.GetDriverByName("ESRI Shapefile")
+
+            # TODO fix filename:
+            temp_shapefile_filename = tempfile.NamedTemporaryFile(
```

Comment:
On a test run of the code, it looks like the following files are not being removed from the `scratch_dir`:

```
test_data_set/scratch_dir/tmpbhirennk.dbf
test_data_set/scratch_dir/tmpbhirennk.prj
test_data_set/scratch_dir/tmpbhirennk.shp
test_data_set/scratch_dir/tmpbhirennk.shx
```

I'm guessing this line is where the `.shp` file is being generated. Could you please update the code so that these four temp files are properly handled and cleaned up, same as the other temp files?

Thank you!

Reply:
Good catch. Thanks!